### PR TITLE
Handle POST In RegisterServlet

### PR DESCRIPTION
**Pull request by hsha0**
Add doPost() function, and tests (1. bad username, 2. new user, 3. existing user)

### CODE REVIEW

File: src/main/webapp/WEB-INF/view/register.jsp [hunk 1]
```diff
@@ -15,6 +15,10 @@
  <div id="container">
    <h1>Register</h1>
 
+   <% if(request.getAttribute("error") != null){ %>
+       <h2 style="color:red"><%= request.getAttribute("error") %></h2>
+   <% } %>
+
```
Comment (TylerSandman) [inline src/main/webapp/WEB-INF/view/register.jsp]: Nit: I realize it's indented like this on the project page, but let's make our best effort to keep indentation consistent (e.g. 2 spaces for all nested elements).

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 2]
```diff
@@ -1,37 +1,70 @@
...
 
- @Override
- public void doGet(HttpServletRequest request, HttpServletResponse response)
-     throws IOException, ServletException {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
 
-   request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
- }
+	request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
```
Comment (TylerSandman) [inline src/main/java/codeu/controller/RegisterServlet.java]: Nit: Here and several other places have tabs. We should not mix tabs and spaces for formatting consistency.

Comment (TylerSandman) [reply src/main/java/codeu/controller/RegisterServlet.java]: Note most editors will let you insert tabs as spaces in settings.

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 3]
```diff
@@ -1,37 +1,70 @@
...
+	String username = request.getParameter("username");
+	String password = request.getParameter("password");
 
-   if (!username.matches("[\\w*\\s*]*")) {
-     request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
-     request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
-     return;
-   }
+	if (!username.matches("[\\w*\\s*]*")) {
+	    request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
```
Comment (TylerSandman) [inline src/main/java/codeu/controller/RegisterServlet.java]: Please keep block indentation consistent (there are 4 spaces here for example, and 2 for setUserStore above)

File: src/test/java/codeu/controller/RegisterServletTest.java [hunk 4]
```diff
@@ -31,4 +31,15 @@ public void testDoGet() throws IOException, ServletException {
...
     }
+    
+    @Test
+    public void testDoPost_BasUsername() throws IOException, ServletException {
+	Mockito.when(mockRequest.getParameter("username")).thenReturn("bad !@#$% username");
+	
+	registerServlet.doPost(mockRequest, mockResponse);
+	
+	Mockito.verify(mockRequest)
+        .setAttribute("error", "Please enter only letters, numbers, and spaces.");
```
Comment (TylerSandman) [inline src/test/java/codeu/controller/RegisterServletTest.java]: Please keep indentation consistent (2 spaces)

**Review by TylerSandman — COMMENTED** [review]
Mostly LGTM with some minor nits. Nice! :)

**Comment by CalamariDude** [comment]
Apologize for late review, 
Tests passing and code looks good aside from the indentions mentioned above^^ :)

**Pull request merged**